conan/tools/ros/rosenv.py

```diff
@@ -0,0 +1,40 @@
+import os
+from conan.api.output import Color
+from conan.tools.env import VirtualBuildEnv, VirtualRunEnv, Environment
+
+
+class ROSEnv(object):
+    """
+    Generator to serve as integration for Robot Operating System 2 development workspaces.
+    It generates a conanrosenv.sh file that when sources sets variables so the Conan
+    dependencies are found by CMake and the run environment is also set.
+
+    IMPORTANT: This generator should be used together with CMakeDeps and CMakeToolchain generators.
+    """
+
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self.variables = {}
+        self._virtualbuildenv = VirtualBuildEnv(conanfile)
+        self._virtualbuildenv.basename = "conanrosenv"
+        self._virtualrunenv = VirtualRunEnv(conanfile)
```

Comment:
```suggestion
        self._virtualbuildenv = VirtualBuildEnv(conanfile, auto_generate=True)
        self._virtualbuildenv.basename = "conanrosenv"
        self._virtualrunenv = VirtualRunEnv(conanfile, auto_generate=True)
```

Is it intended that this generator inhibits the generation of ``VirtualBuildEnv`` and ``VirtualRunEnv``?